mava/wrappers/gym.py

```diff
@@ -0,0 +1,312 @@
+        Args:
+            env (gym.env): gym env instance.
+            use_shared_rewards (bool, optional): Use individual or shared rewards.
+            Defaults to False.
+            add_global_state (bool, optional) : Create global observations. Defaults to False.
+        """
+        super().__init__(env)
+        self._env = env
+        self.use_shared_rewards = use_shared_rewards
+        self.add_global_state = add_global_state
+        self.num_agents = len(self._env.action_space)
+        self.num_actions = self._env.action_space[0].n
+
+    def reset(
+        self, seed: Optional[int] = None, options: Optional[dict] = None
+    ) -> Tuple[np.ndarray, Dict]:
+
+        if seed is not None:
+            self.env.seed(seed)
+
+        agents_view, info = self._env.reset()
+
+        info = {"actions_mask": self.get_actions_mask(info)}
+        if self.add_global_state:
+            info["global_obs"] = self.get_global_obs(agents_view)
+
+        return np.array(agents_view), info
+
+    def step(self, actions: NDArray) -> Tuple:
+
+        agents_view, reward, terminated, truncated, info = self._env.step(actions)
+
+        info = {"actions_mask": self.get_actions_mask(info)}
+        if self.add_global_state:
+            info["global_obs"] = self.get_global_obs(agents_view)
+
+        if self.use_shared_rewards:
+            reward = np.array([np.array(reward).sum()] * self.num_agents)
+        else:
+            reward = np.array(reward)
+
+        return agents_view, reward, terminated, truncated, info
+
+    def get_actions_mask(self, info: Dict) -> NDArray:
+        if "action_mask" in info:
+            return np.array(info["action_mask"])
+        return np.ones((self.num_agents, self.num_actions), dtype=np.float32)
+
+    def get_global_obs(self, obs: NDArray) -> NDArray:
+        global_obs = np.concatenate(obs, axis=0)
+        return np.tile(global_obs, (self.num_agents, 1))
+
+
+class GymLBFWrapper(gym.Wrapper):
+    """Wrapper for rware gym environments"""
+
+    def __init__(
+        self,
+        env: gym.Env,
+        use_shared_rewards: bool = False,
+        add_global_state: bool = False,
+    ):
+        """Initialize the gym wrapper
+        Args:
+            env (gym.env): gym env instance.
+            use_shared_rewards (bool, optional): Use individual or shared rewards.
+            Defaults to False.
+            add_global_state (bool, optional) : Create global observations. Defaults to False.
+        """
+        super().__init__(env)
+        self._env = env  # not having _env leaded tp self.env getting replaced --> circular called
+        self.use_shared_rewards = use_shared_rewards
+        self.add_global_state = add_global_state  # todo : add the global observations
+        self.num_agents = len(self._env.action_space)
+        self.num_actions = self._env.action_space[
+            0
+        ].n  # todo: all the agents must have the same num_actions, add assertion?
```

Comment:
Did you ever implement that assertion? Otherwise maybe a warning saying that in LBF with assume agents are homogeneous.

Reply:
There's a built-in check in gym's async workers. I removed the comment.